src/main/java/frc/robot/commands/GoToPosePID.java

```diff
@@ -0,0 +1,46 @@
+
+package frc.robot.commands;
+
+import java.util.function.Supplier;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Drivetrain;
+
+/**
+ * Tests the odometry of the robot by driving a certain distance and calculating the error.
+ */
+public class GoToPosePID extends CommandBase {
+
+  private Drivetrain m_drive; 
+  
+  private Supplier<Pose2d> m_pose;
+  
+  public GoToPosePID(Drivetrain drive, Supplier<Pose2d> pose) {
+    m_drive = drive; 
+    // finalPose is position after robot moves from current position-- startPose-- by the values that are inputted-- distanceToMove
+    m_pose = pose;
+    
+    addRequirements(drive);
+}
+  
+@Override
+public void initialize() {
+}
+
+@Override
+public void execute() {
+  m_drive.runChassisPID(m_pose.get().getX(), m_pose.get().getY(), m_pose.get().getRotation().getRadians()); 
```

Comment:
there will be a method in the drivetrain cleanup PR that acepts a pose that should make this line of code much more compact and easier to fallow. Just something to note when the PR get merged into main.